spec/services/github_service.rb

```diff
@@ -0,0 +1,130 @@
+require 'rails_helper'
+
+RSpec.describe GithubService do
+  subject { described_class.call(payload: payload, event: event) }
+
+  describe 'events' do
+    context 'pull request' do
+      let!(:payload) { (create :pull_request_payload).merge((create :repository_payload)) }
```

Comment:
This line

`let!(:payload) { (create :pull_request_payload).merge((create :repository_payload)) }`

got me confused.

The reason is that when reading it I thought there was a logical error in merging the attributes of a pull_request with the attributes of a repository.

Going back to the factory source code I saw that the merge was ok because `create :pull_request_payload` returns 

```
pull_request: {
  id: ...
  ...
}
repository: {
  id: ...
}
```

That made me think that the factory should not include the first payload key but only the attributes:

`create :pull_request_payload`

`{
  id: ...
}`

The reason is that it would allow to insert a payload object into another payload object using a different name.

Here is an example:

If `build :user_payload` returns `{ id: ... }` instead of `{ user: { id... }` it allows to use like this:

```
pull_request_event = {
   action: ...,
   pull_request: {
      user: (build :user, id: 1)
   },
   requested_reviewer: (build :user, id: 2)
}
```

Including only the attributes of a payload object would also be consistent with the behaviour of FactoryBot when it builds a model object.

Reply:
I will refactor the payload factories to have a sub factory with an after_create which will merge the requested payload. Here an example:
```
    factory :pull_request_payload_with_repository do
      after(:create) do |pull_request_payload|
        pull_request_payload.merge!((create :repository_payload))
      end
    end
```

Reply:
How about having 2 different factories, one for a `pull_request_event_payload` and another one for a `pull_request_payload` (the PR object, not the PR event).

The same for `repository_event_payload` and `repository_payload`.

After all the definition of the Webhooks treats this concepts like composable objects.

This would allow to use composition of the events and objects like it does with other attributes:

```
# default pull request event that includes a default pull request and a default repository
build :pull_request_event_payload
```

```
# override the pull request event with a different repository
build : pull_request_event_payload, repository: (build :repository_payload)
```

```
# override of the pull request event with a different pull request and a different repository
build : pull_request_event_payload, pull_request: (build :pull_request_payload), repository: (build :repository_payload)
```